nemo/core/classes/mixins/adapter_mixins.py

```diff
@@ -159,6 +173,11 @@ def add_adapter(self, name: str, cfg: DictConfig):
             cfg: A DictConfig or Dataclass that contains at the bare minimum `__target__` to instantiate a
                 new Adapter module.
         """
+        if self.get_accepted_adapters():
+            assert (
```

Comment:
Dont assert, check with if and raise appropriate error.

Reply:
This check is also incorrect. Name should be name of the class, not some random name.
Adapters can have any name, including module adapter style names ("module:adapter_name").

You can check class via class inflation (look at model_utils.import_class_from_path) or via str match of class, but not of individual adapter names.

Reply:
Hmm does using model_utils.import_class_from_path() here cause circular dependency? If not, might as well load the class and do an issubclass() check rather than explicit str check. 

Subclasses can then be supported, plus if paths change in the future due to refactor, older checkpoints won't break (since we can do import in old dir init files).